### DISCUSSION TITLE
Generate imodel object

### DISCUSSION THREAD

**Discussion by summer-shit** [iTwin/itwinjs-core#2176] [Q&A] [answered]
Recently started to understand imodel.js. Can't I directly read DGN, DWG and other files locally to generate imodel objects?

Is. the  BIM file format that Bently generates separately for imodel.js? In other words, what is the source of the. BIM file? First contact with this format

**Comment by abeesh** [comment, marked as answer]
You will need an iModel connector to ingest files like dgn, dwg, revit etc.. For your local development, please see this section on how to get started. https://www.itwinjs.org/learning/tutorials/create-test-imodel-offline/

For working with data on iModelHub, please see https://www.bentley.com/en/products/product-line/digital-twins/itwin-synchronizer

The [synchronization wiki](https://communities.bentley.com/products/digital-twin-cloud-services/itwin-services/w/synchronization-wiki)  provides more details.
